internal/pkg/gateway/commit/blocknotifier.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type blockListener interface {
+	receiveBlock(*ledger.CommitNotification)
+	close()
+}
+
+type blockNotifier struct {
+	commitChannel <-chan *ledger.CommitNotification
+	done          <-chan struct{}
+	lock          sync.Mutex
+	listeners     []blockListener
+	closed        bool
```

Comment:
nit: convention - lock just above what it protects (in other file as well)
```suggestion
	listeners     []blockListener
	
	lock          sync.Mutex
	closed        bool
```